### ISSUE TITLE
[1.7.1][iOS] Cannot interact with Notifications Alert

### ISSUE THREAD

**Issue by welnanick** [appium/appium#9370]
## The problem

The app I am testing requests the permission to send the user notifications, but now appium cannot locate the buttons from the alert. appium-desktop shows it has an accessibility id, and a unique xpath, but neither work to locate the element. I have also tried an iOS Predicate string which also does not work.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.7.1
* Last Appium version that did not exhibit the issue (if applicable): 1.7.0 Beta
* Desktop OS/version used to run Appium: macOS Sierra 10.12.6
* Node.js version (unless using Appium.app|exe): 8.5.0
* Mobile platform/version under test: iOS 10.3.2
* Real device or emulator/simulator: Real Device
* Appium CLI or Appium.app|exe: CLI

## Details

I've tried the following: accessibility id: Allow, xpath: //XCUIElementTypeButton[@name="Allow"], and predicate string: type == 'XCUIElementTypeButton' AND name =='Allow'. The accessibility id and xpath are the ones appium-desktop shows.

I can downgrade to 1.7.0 and check if it works there, to see if the bug was introduced in the transition from 1.7.0 beta -> 1.7.0 or from 1.7.0 -> 1.7.1 if you would like me to. **_My iOS testing skipped from 1.7.0 Beta to 1.7.1, so I have not tested against 1.7.0 with iOS_**.

## Link to Appium logs
[logs](https://gist.github.com/welnanick/d096b09aac388853dc2e4a1230a6a8ea)

**Comment by mykola-mokhnach** [comment]
You should either use Alert interface (`driver.switchTo().alert()`) or [mobile: alert](https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/ios-xctest-mobile-gestures.md) endpoint.

**Comment by welnanick** [comment]
Alright, sorry for opening the bug. Odd that it used to work and now it doesn't, but I suppose that is the life of not using the proper method. Thanks for the quick reply!
